benchmarks/newton_solver_benchmark_set/spiegelman_et_al_2016/input.prm

```diff
@@ -0,0 +1,141 @@
+set Nonlinear solver tolerance             = 1e-14
+set Pressure normalization                 = no
+
+subsection Solver parameters
+  subsection Stokes solver parameters
+    set Number of cheap Stokes solver steps = 0
+    set Linear solver tolerance             = 1e-5
+  end 
+  subsection Newton solver parameters
+    set Max pre-Newton nonlinear iterations      = 5
+    set SPD safety factor                        = 0.9
+    set Nonlinear Newton solver switch tolerance = 1e-20 #  
+    set Max Newton line search iterations        = 0
+    set Maximum linear Stokes solver tolerance   = 9e-1
+    set Use Newton residual scaling method       = false
+    set Use Newton failsafe                      = false
+    set Stabilization preconditioner             = SPD
+    set Stabilization velocity block             = SPD
+  end
+end
+
+subsection Boundary temperature model
+  set Model name = box
+end
+
+
+ subsection Initial temperature model
+   set Model name = function
+ end
+
+ subsection Initial composition model
+   set Model name = function
+   subsection Function
+    set Variable names      = x,z
+    set Function constants  = H = 30e3
+    # The function below lays out the composition structure as described in the Spiegelman paper. 
+    # It generates a composition at the bottom wich has a square with round eges sticking out in 
+    # the center.
+    set Function expression = if(z < 0.25 * H,1, \
+if(z < (0.25+(1/12)-0.02)*H & x > (2-(1/12))*H & x < (2+(1/12))*H,1, \
+if(z < (0.25+(1/12))*H & (x > (2-(1/12)+0.02)*H & x < (2+(1/12)-0.02)*H | (x-(2-(1/12)+0.02)*H)^2 + (z-(0.25+(1/12)-0.02)*H)^2 < (0.02*H)^2 | (x-(2+(1/12)-0.02)*H)^2 + (z-(0.25+(1/12)-0.02)*H)^2 < (0.02*H)^2),1, \
+if(z < (0.25+0.02)*H && x > (2-(1/12)-0.02)*H && x < (2+(1/12)+0.02)*H && ((x-(2-(1/12)-0.02)*H)^2 + (z-(0.25+0.02)*H)^2 > (0.02*H)^2 && (x-(2+(1/12)+0.02)*H)^2 + (z-(0.25+0.02)*H)^2 > (0.02*H)^2),1,0))))
+   end
+ end
+
+
+subsection Gravity model
+  set Model name = vertical
+  subsection Vertical
+    set Magnitude = 9.81
+  end
+end
+
+
+subsection Geometry model
+  set Model name = box
+
+  subsection Box
+    set X extent      = 120e3
+    set Y extent      = 30e3
+    set X repetitions = 4
+  end
+end
+
+
+subsection Compositional fields
+  set Number of fields = 1
+#parameters for all rheologies
+      set Conductivities                         = 2.25,          2.25
+      set Heat capacities                        = 1250,          1250
+      set Reference temperatures                 = 0,             0
+      set Reference densities                    = 2700.0,        2700.0
+      #plastic parameters
+      set Cohesions                              = 1e8,0
+      set Angles of internal friction            = 30,0
+      #Dislocation parameters
+      set Stress exponents                       = 50,            1
+      set Initial viscosities                    = 1e20, 1e21
+      set Viscous prefactors                     = 39905246.2994, 1e21
```

Comment:
I'm still not sure exactly why the `Viscous prefactors` need to be defined, as they just represent a fixed viscosity value and are not part of an equation for diffusion or dislocation creep. Specifically, what does `39905246.2994` represent? Can you remove this variable and only use `Initial viscosities` instead? In the .cc file, it looks if no regularization is used, the viscosity is set to those defined in `Viscous prefactors`. If a different fixed viscosity needs to be defined than  `Viscous prefactors`, please add documentation that specifies the difference both where the variable is declared and in the documentation at the base of the .cc file.

Reply:
The value `39905246.2994` is not used at all since it is in the stop of composition 0, which is ignored. I could give it any value, but I guess I should change it to something shorting an so that it is clear that it is insensible. 0? I will add that as documentation to the input file.

I could use initial viscosities, since they do not seem to be used in any place, but I don't think a initial viscosity as a name is appropriate for representing a constant viscosity. The prefactor is a bit better since it a place where at least I put constant viscosities. I agree that it is not perfect, but since I don't think material model is not going to be used for any practical purposes, since it is specifically tailored for this problem, I doubt it is worth spending much time in finding a perfect name for this. I have added the disclaimer on that at the top of the cc file.